### PR TITLE
FFI: add function rnp_output_to_file()

### PR DESCRIPTION
**Pull request by ni4**
Main goal of this PR was to add rnp_output_t creation functions which would allow to control file overwrite/random temporary file creation. 
Probably it should have another name, like rnp_output_to_path_ex() or so on.
However then we'll probably need to control temporary dir creation as well.

### CODE REVIEW

File: src/tests/support.cpp [hunk 1]
```diff
@@ -103,6 +103,19 @@ file_contents(const char *path, ssize_t *size)
     return mem;
 }
 
+int
```
Comment (dewyatt) [inline src/tests/support.cpp]: I know you're just moving this code but I think this should be `off_t`. I know on my system `st_size` is 8 bytes and int is 4.

Comment (ni4) [reply src/tests/support.cpp]: @dewyatt  Agree on this. Just updated the PR.

**Comment by ronaldtse** [comment]
@dewyatt as you mentioned about release scheduling, please feel free to merge at the right time. Thanks!

**Comment by dewyatt** [comment]
Should be fine to merge now, I'm just trying to consider 0.13 as mostly frozen so I probably won't cherry-pick this into that branch.

**Pull request merged**